examples/mugen/retrieval/s3d.py

```diff
@@ -111,262 +111,42 @@ def forward(self, x):
-            SepConv3d(96, 208, kernel_size=3, stride=1, padding=1),
+class Branch1or2(nn.Sequential):
+    def __init__(self, in_planes, mid_planes, out_planes):
+        super(Branch1or2, self).__init__(
+            BasicConv3d(in_planes, mid_planes, kernel_size=1, stride=1),
+            SepConv3d(mid_planes, out_planes, kernel_size=3, stride=1, padding=1),
         )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(480, 16, kernel_size=1, stride=1),
-            SepConv3d(16, 48, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
-            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(480, 64, kernel_size=1, stride=1),
-        )
-
-    def forward(self, x):
-        x0 = self.branch0(x)
-        x1 = self.branch1(x)
-        x2 = self.branch2(x)
-        x3 = self.branch3(x)
-        out = torch.cat((x0, x1, x2, x3), 1)
-        return out
-
 
-class Mixed4c(nn.Module):
-    def __init__(self):
-        super(Mixed4c, self).__init__()
 
-        self.branch0 = nn.Sequential(
-            BasicConv3d(512, 160, kernel_size=1, stride=1),
-        )
-        self.branch1 = nn.Sequential(
-            BasicConv3d(512, 112, kernel_size=1, stride=1),
-            SepConv3d(112, 224, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(512, 24, kernel_size=1, stride=1),
-            SepConv3d(24, 64, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
+class Branch3(nn.Sequential):
+    def __init__(self, in_planes, out_planes):
+        super(Branch3, self).__init__(
             nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(512, 64, kernel_size=1, stride=1),
+            BasicConv3d(in_planes, out_planes, kernel_size=1, stride=1),
         )
 
-    def forward(self, x):
-        x0 = self.branch0(x)
-        x1 = self.branch1(x)
-        x2 = self.branch2(x)
-        x3 = self.branch3(x)
-        out = torch.cat((x0, x1, x2, x3), 1)
-        return out
 
-
-class Mixed4d(nn.Module):
-    def __init__(self):
-        super(Mixed4d, self).__init__()
-
-        self.branch0 = nn.Sequential(
-            BasicConv3d(512, 128, kernel_size=1, stride=1),
-        )
-        self.branch1 = nn.Sequential(
-            BasicConv3d(512, 128, kernel_size=1, stride=1),
-            SepConv3d(128, 256, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch2 = nn.Sequential(
-            BasicConv3d(512, 24, kernel_size=1, stride=1),
-            SepConv3d(24, 64, kernel_size=3, stride=1, padding=1),
-        )
-        self.branch3 = nn.Sequential(
-            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
-            BasicConv3d(512, 64, kernel_size=1, stride=1),
-        )
+class MixedConvsBlock(nn.Module):
+    def __init__(self, in_planes, b0_out, b1_mid, b1_out, b2_mid, b2_out, b3_out):
```

Comment:
similar comment here around naming

Reply:
NIT: just a suggestion on the parameters, we might consider to single parameter called: `branch_layers` and it is a list of list of integers representing the layer parameters of the branch, i.e:
```
[[b0_out], [b1_mid, b1_out], [b2_mid, b2_out], [b3_out]]
```
with this the constructor can become: `def __init__(self, in_planes, branch_layers)`

Reply:
changed the param format to `List[List[int]]`, but keeping the "mid" naming for consistency with torchvision [video resnet](https://pytorch.org/vision/stable/_modules/torchvision/models/video/resnet.html) naming